src/main/scala/firrtl/backends/verilog/PadWidthsVerilog.scala

```diff
@@ -0,0 +1,51 @@
+// See LICENSE for license details.
+
+package firrtl.backends.verilog
+
+import firrtl.ir._
+import firrtl.PrimOps._
+import firrtl.Mappers._
+import firrtl.{Dshlw, SystemVerilogEmitter, Transform, VerilogEmitter, bitWidth}
+import firrtl.options.Dependency
+import firrtl.stage.Forms
+import firrtl.stage.TransformManager.TransformDependency
+
+
+/** Adds padding for mul, div, rem, dshl which breaks firrtl width invariance, but is nmeeded to match Verilog semantics */
+object PadWidthsVerilog extends firrtl.passes.Pass {
+
+  override def prerequisites: Seq[TransformDependency] = Forms.MidForm ++
+    Seq( Dependency(firrtl.passes.LowerTypes),
+         Dependency(firrtl.transforms.RemoveReset),
+         Dependency[firrtl.transforms.RemoveWires] )
+
+  override def optionalPrerequisites = Seq(Dependency[firrtl.transforms.ConstantPropagation])
```

Comment:
```suggestion
```
Given the issue with ConstProp in the Verilog emitter, I would not include this optional Prereq. Rather, some aspects of PadWidths should be a different transform that still runs before ConstProp

Reply:
Yeah, makes sense. I am currently debugging this failure and it seems related to the ConstantPropagation invalidation as well as Legalize. I will try and understand what `Legalize` is actually supposed to do and why PadWidths would invalidate it.

Reply:
no idea about why `PadWidths` invalidates `Legalize`, but the purpose of `Legalize` was (originally) to be "const prop-like things that are required for correct Verilog emission". This responsibility has clearly been distributed all over so that distinction no longer exists, but that's the kind of functionality in their.

Reply:
I worked through all the cases that could be invalidated by `PadWidths` and I am like 70% sure that with the two calls to const prop functions in the latest commit Legalize should not be invalidated.

Reply:
The one case that it used to invalidate was when padding literals: `UInt<1>(0)` --PadWidths--> `pad(UInt<1>(0), 4)` --Legalize/ConstProp--> `UInt<4>(0)`.

Reply:
There are others, but this is the most obvious one.